menuinst/win32.py

```diff
@@ -251,26 +254,28 @@ def remove(self):
 
     def create(self, remove=False):
         args = []
+        # cmd is our root installation interpreter
+        cmd = join(unicode_prefix, u"pythonw.exe").replace("\\", "/")
+        # each of these roll the subprocess cmd into args
         if "pywscript" in self.shortcut:
-            cmd = join(self.menu.prefix, u"pythonw.exe").replace("\\", "/")
+            subprocess_cmd = join(self.menu.prefix, u"pythonw.exe").replace("\\", "/")
             args = self.shortcut["pywscript"].split()
-            args = get_python_args_for_subprocess(self.menu.prefix, args, cmd)
+            args = get_python_args_for_subprocess(self.menu.prefix, args, subprocess_cmd)
         elif "pyscript" in self.shortcut:
-            cmd = join(self.menu.prefix, u"python.exe").replace("\\", "/")
+            subprocess_cmd = join(self.menu.prefix, u"python.exe").replace("\\", "/")
             args = self.shortcut["pyscript"].split()
-            args = get_python_args_for_subprocess(self.menu.prefix, args, cmd)
+            args = get_python_args_for_subprocess(self.menu.prefix, args, subprocess_cmd)
         elif "webbrowser" in self.shortcut:
-            cmd = join(unicode_prefix, u'pythonw.exe')
             args = ['-m', 'webbrowser', '-t', self.shortcut['webbrowser']]
```

Comment:
Since the `"system"` case below is now changed to use `cwp`, the same could be done here which in effect makes
```python
args = get_python_args_for_subprocess(self.menu.prefix, args, subprocess_cmd)
```
be done in any case and can thus be done after the if-elif-else-clause to reduce duplicate code :)

Reply:
I have been reviewing these changes and I don't get what's going on here (not sure if I'm referring to the changes in the PR or some of the stuff we've done previously).

We run both `script` and `system` commands through `pythonw` and `cwp` now which seems wrong to me. We have `script` that contains:
```
        "script": "${PREFIX}/Library/bin/bash.exe",
```
and:
```
        "script": "${PREFIX}/Library/bin/rstudio.exe",
```
.. why would either of these want to be run through `pythonw` and `cwp.py`?

For the `Anaconda Prompt` we have `system` set as:
```
            "system": "%windir%\\System32\\cmd.exe",
            "scriptarguments": ["/K", "${ROOT_PREFIX}\\Scripts\\activate.bat", "${PREFIX}"],
```
.. and again we run that through `pythonw` and `cwp` now!